### PR TITLE
Parallel simulations

### PR DESCRIPTION
**Pull request by rouson**
With this pull request,
* Vegetables generates the integration test driver.
* CTest executes the integration test directly rather than
      running a shell script.
* The Sourcery test oracle evaluatees the test test as passing
    or failing in Fortran with no need to compare external files.
* Only image 1 reads the input file.
* Image 1 broadcasts the input data to all other images.
    
 To Do:
1. Construct a reference object from a reference implemenation.
2. Restrict computation on each image to its data partition.
3. Gather the result or invoke Sourcery's co_all to verify
     test passing globally.
4. Have image 1 report the result.

### CODE REVIEW

**Comment by everythingfunctional** [comment]
By making those functions elemental and removing the loop over simulations, aren't you now effectively doing the loop over `nsim` multiple times? Don't you still want the explicit loop over `nsim`, but split up the iterations across images?

**Comment by rouson** [comment]
@everythingfunctional just to memorialize our discussion of the above comment: while it's true that the new array statements are equivalent to serial loops, there's no guarantee that the  compiler sets up loops to do the array work. Using array statements more clearly communicates (to the reader and the compiler) the implicit lack of ordering, which the compiler might then exploit through one or more various means: vectorization, parallelization, or offloading.

File: README.md [hunk 1]
```diff
@@ -33,27 +33,30 @@ Earlier versions might work.
...
+-----------------
+To obtain miniFAVOR, execute the following command:
 
-Cloning FAVOR
-------------------
-* To build miniFAVOR from source, clone this repository,
-  execute the following command at a command line:
 ```
-git clone git@github.com:everythingfunctional/miniFAVOR.git
+git clone --recursive git@github.com:everythingfunctional/miniFAVOR.git
```
Comment (everythingfunctional) [inline README.md:45]:
```suggestion
git clone --recursive git@github.com:sourceryinstitute/miniFAVOR.git
```

File: app/miniFAVOR.f90 [hunk 2]
```diff
@@ -14,98 +14,90 @@
...
-      integer, parameter :: nmaterials=2
-
-      associate(content => reshape([material_content%Cu(),material_content%Ni()], [nsim, nmaterials] ))
-        call write_OUT(fn_IN, n_OUT, n_DAT, &
-          a, b, nsim, ntime, details, Cu_ave, Ni_ave, Cu_sig, Ni_sig, fsurf, RTndt0, &
-          R_Tndt, CPI, CPI_avg, K_hist, content, Chemistry_factor)
+        associate( &
+          material_content => &
+          material_content_t(input_data%Cu_ave(), input_data%Ni_ave(), input_data%Cu_sig(),input_data%Ni_sig(), samples) &
+        )
```
Comment (everythingfunctional) [inline app/miniFAVOR.f90:72]:
```suggestion
        associate(material_content => material_content_t( &
            input_data%Cu_ave(),  input_data%Ni_ave(), input_data%Cu_sig(), input_data%Ni_sig(), samples))
```

File: .gitmodules [hunk 3]
```diff
@@ -12,4 +12,4 @@
 	url = https://gitlab.com/everythingfunctional/parff
 [submodule "src/git-submodules/sourcery"]
 	path = src/git-submodules/sourcery
-	url = https://github.com/sourceryinstitute/sourcery
+	url = git@github.com:sourceryinstitute/sourcery
```
Comment (everythingfunctional) [inline .gitmodules:15]:
```suggestion
	url = https://github.com/sourceryinstitute/sourcery
```

That'll fix the current failure

**Comment by rouson** [comment]
@everythingfunctional the CI test failures appear to be related to an actual MPI issue. Maybe we should have the CI script build MPICH from source and then use it to build OpenCoarrays.  I generally find MPICH more robust and easier to use than OpenMPI.

**Pull request merged**